documentation/advanced/tutorial-connection-indicator.asciidoc

```diff
@@ -0,0 +1,207 @@
+connectionIndicator.expandedDuration = 2000; // 2000 ms is the default
+// The message for the “connected” connection state
+connectionIndicator.onlineText = 'Online';
+// The message for the “connection lost” state
+connectionIndicator.offlineText = 'Connection lost';
+// The message for the “reconnecting” connection state
+connectionIndicator.reconnectingText = 'Connection lost, trying to reconnect...';
+----
+
+== Displaying a Modal Curtain
+
+To show an alternative for the default connection indicator theme, this examples demonstrates how to show a
+loading state that darkens the UI. The darkening is animated, so that it does not flash
+the screen. The darkening starts after the server side round-trip takes over 0.5
+seconds (300 ms configuration delay + 200 ms animation delay).
+
+[NOTE]
+Apart from adding the CSS, the default theme should be explicitly disabled via TypeScript,
+as shown in the previous chapter.
+
+[source,css]
+----
+.v-loading-indicator {
+  position: fixed; /* Occupy whole screen even if scrolled */
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  /* Eat mouse events when visible, doesn't prevent scrolling */
+  pointer-events: auto;
+  /* Animated with delay to prevent flashing */
+  animation: fadein 0.3s ease-out 0.2s normal 1 both;
+  z-index: 2147483647;
+}
+@keyframes fadein {
+  0% {
+    background: rgba(0,0,0,0);
+  }
+  100% {
+    background: rgba(0,0,0,.5); /* Darkens the UI */
+  }
+}
+----
+
+The next image illustrates an application with the modal curtain visible during loading
+(above) compared to the normal state (below).
+
+image:images/modal_curtain.png[modal curtain,700,700]
+
+== Displaying a Changing Loading Indicator
+
+Once the loading indicator is displayed, it gets the class name `first`. After
+the second and third configurable delays, it gets the `second` and the `third`
+class names respectively. You can use those class names in your styling to let
+the look reflect how long time the user has been waiting.
+
+The following style snippet demonstrates how to create an animation that changes
+color as the user is waiting.
+
+[NOTE]
+Apart from adding the CSS, the default theme should be explicitly disabled via TypeScript.
+
+[source,css]
+----
+.v-loading-indicator {
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  pointer-events: auto;
+  z-index: 2147483647;
+}
+.v-loading-indicator:before {
+  width: 76px;
+  height: 76px;
+
+  position: absolute;
+  top: 50%;
+  left: 50%;
+
+  margin: -38px 0 0 -38px;
+
+  border-radius: 100%;
+  animation: bouncedelay 1.2s infinite 0.4s ease-in-out both;
+  content: "";
+}
+
+.v-loading-indicator.first:before {
+  background-color: skyblue;
+}
+
+.v-loading-indicator.second:before {
+  background-color: salmon;
+}
+
+.v-loading-indicator.third:before {
+  background-color: red;
+}
+
+@keyframes bouncedelay {
+  0%, 80%, 100% {
+    transform: scale(0);
+  } 40% {
+    transform: scale(1.0);
+  }
+}
+----
```

Comment:
The same as above, would be helpful to also show where to apply these styles.

Reply:
Also added a link above the example.